mmv1/third_party/terraform/tests/resource_dataplex_task_test.go

```diff
@@ -0,0 +1,177 @@
+)
+
+func TestAccDataplexTaskDataplexTaskPrimary_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"project_name":  acctest.GetTestProjectFromEnv(),
+		"random_suffix": RandString(t, 10),
+	}
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckDataplexTaskDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataplexTask_dataplexTaskPrimary(context),
+			},
+			{
+				ResourceName:            "google_dataplex_task.example",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "lake", "task_id"},
+			},
+			{
+				Config: testAccDataplexTask_dataplexTaskPrimaryUpdate(context),
+			},
+			{
+				ResourceName:            "google_dataplex_task.example",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "lake", "task_id"},
+			},
+		},
+	})
+}
+
+func testAccDataplexTask_dataplexTaskPrimary(context map[string]interface{}) string {
+	return tpgresource.Nprintf(`
+data "google_project" "project" {
+
+}
+
+resource "google_dataplex_lake" "example" {
+  name         = "tf-test-lake%{random_suffix}"
+  location     = "us-central1"
+  project = "%{project_name}"
+}
+
+
+resource "google_dataplex_task" "example" {
+
+    task_id      = "tf-test-task%{random_suffix}"
+    location     = "us-central1"
+    lake         = google_dataplex_lake.example.name
+    trigger_spec  {
+        type = "ON_DEMAND"
+    }
+    
+    execution_spec {
+        service_account = "${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+    }
+    
+    spark {
+        python_script_file = "gs://dataproc-examples/pyspark/hello-world/hello-world.py"
+    }
+    
+    project = "%{project_name}"
+    
+}
+`, context)
+}
+
+func testAccDataplexTask_dataplexTaskPrimaryUpdate(context map[string]interface{}) string {
+	return tpgresource.Nprintf(`
+data "google_project" "project" {
+
+}
+
+resource "google_dataplex_lake" "example" {
+  name         = "tf-test-lake%{random_suffix}"
+  location     = "us-central1"
+  project = "%{project_name}"
+}
+
+
+resource "google_dataplex_task" "example" {
+
+    task_id      = "tf-test-task%{random_suffix}"
+    location     = "us-central1"
+    lake         = google_dataplex_lake.example.name
+    trigger_spec  {
+        type = "ON_DEMAND"
+    }
+    
+    execution_spec {
+        service_account = "${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+    }
+    
+    spark {
+        python_script_file = "gs://dataplex-clouddq-api-integration-test/clouddq_pyspark_driver.py"
+    }
+    
+    project = "%{project_name}"
+    
+}
+`, context)
+}
+
+func testAccCheckDataplexTaskDestroyProducer(t *testing.T) func(s *terraform.State) error {
```

Comment:
```suggestion
```

this function does not need to be included in this test, it is already created downstream via the generated test